### PR TITLE
Moved metadata_store into database and content_discovery

### PR DESCRIPTION
**Pull request by qstokkink**
Related to #7669, following the plan of https://github.com/Tribler/tribler/issues/7669#issuecomment-1849549637

The changes of this PR are simply moves and dead code removal.

Unit tests have been added to cover missing lines (in our existing code). The lines in `database_component.py` that are part of `gui_test_mode` or `chant_testnet` are not covered.

### CODE REVIEW

File: src/tribler/core/components/database/database_component.py [hunk 1]
```diff
@@ -1,23 +1,67 @@
...
-        db_path = self.session.config.state_dir / STATEDIR_DB_DIR / "tribler.db"
-        if self.session.config.gui_test_mode:
+        config = self.session.config
+        db_path = config.state_dir / STATEDIR_DB_DIR / "tribler.db"
+        if config.gui_test_mode:
             db_path = ":memory:"
 
         self.db = TriblerDatabase(str(db_path))
 
+        # TODO: merge the code below into the TriblerDatabase
```
Comment (drew2a) [inline src/tribler/core/components/database/database_component.py:26]: Is this a TODO for the next PR, or is it a forgotten TODO for the current PR?

**Review by drew2a — APPROVED** [review]
👍

File: src/tribler/core/components/database/database_component.py (hunk 1 above)
Comment (qstokkink) [reply src/tribler/core/components/database/database_component.py:26]: This is for the next PR.

Comment (drew2a) [reply src/tribler/core/components/database/database_component.py:26]: Some of the developers and I in this group consider it bad, or let's say "not good," practice to merge changes with TODOs into the main branch.

I'm not stopping you from doing this by not giving acceptance to the PR. I'm just highlighting that there are many points of view on this practice, some of which consider it to be a bad practice that leads to code quality degradation.

https://www.reddit.com/r/learnprogramming/comments/r25bj7/are_is_committing_code_to_github_with_todos_bad/?rdt=46532

Comment (qstokkink) [reply src/tribler/core/components/database/database_component.py:26]: I agree with them. However, our current linters are configured to allow this. Notably, I did not have to disable any warnings to pass our PR checks.

Once we fix https://github.com/Tribler/tribler/issues/1722 we can adopt a more strict stance on this.

Comment (drew2a) [reply src/tribler/core/components/database/database_component.py:26]: Are you saying that if you understand there is a bad practice, but there are no configured linter issues, then it is okay for you to commit and merge the code despite its quality?

Comment (qstokkink) [reply src/tribler/core/components/database/database_component.py:26]: Even though I disagree with the practice personally, this is the approach that the team has chosen and I adapt my workflow to it.

If I were the only developer, I would make make own standards and conform to what I would like to see. Sadly, I am not the only developer and I have to conform to the existing rules and linter settings.

**Pull request merged**